### ISSUE TITLE
Joystick teleop control

### ISSUE THREAD

**Issue by jrggaldamez** [ros-planning/moveit#1189]
### Description

Hello everyone.
So I wanted to use my PS3 controller in order to move my robot through Rviz / MoveIt!. I followed the steps on the MoveIt tutorial for this. But when it came to launching the joystick_control.launch file I get an issue with the moveit_joy.py apparently. My error is as following:
[FATAL] [1541427087.528021]: Check if you started movegroups. Exiting.
Traceback (most recent call last):
  File "/home/programmer/ros_ws/src/moveit/moveit_ros/visualization/scripts/moveit_joy.py", line 42, in <module>
    app = MoveitJoy()
  File "/home/programmer/ros_ws/src/moveit/moveit_ros/visualization/src/moveit_ros_visualization/moveitjoy_module.py", line 383, in __init__
    self.updatePlanningGroup(0)
  File "/home/programmer/ros_ws/src/moveit/moveit_ros/visualization/src/moveit_ros_visualization/moveitjoy_module.py", line 407, in updatePlanningGroup
    raise rospy.ROSInitException(msg)
rospy.exceptions.ROSInitException: Check if you started movegroups. Exiting.
[moveit_joy-2] process has died [pid 17808, exit code 1, cmd /home/programmer/ros_ws/src/moveit/moveit_ros/visualization/scripts/moveit_joy.py __name:=moveit_joy __log:=/home/programmer/.ros/log/75fd38d0-e0f9-11e8-9bf8-484d7ed47ff1/moveit_joy-2.log].
log file: /home/programmer/.ros/log/75fd38d0-e0f9-11e8-9bf8-484d7ed47ff1/moveit_joy-2*.log

Basically meaning it cannot access the move groups of my robot. I am using the Sawyer robot from Rethink Robotics. I also created manually a move group through the MoveIt setup assistant. I launched the demo file and with this I was able to control the simulation using the PS3 controller.

Another issue I have encountered is that whilst trying to use the MoveIt Commander Command line script, it is not able to find the move group either. I ran a test python file to see if the command "get_group_names()" would be able to detect it and it did. The group should be called 'right_arm'. So I am wondering if anyone has has any similar experiences in this area. I would be really thankful for any advice.

### Your environment
* ROS Distro: [|Kinetic]
* OS Version: e.g. Ubuntu 16.04

**Comment by azhural** [comment]
Trying to answer a question on ROS answers I ran into this issue with 2 robot setups. A UR5 setup for work and the fanuc setup from bezier examples. I then had a look at the [moveitjoy_module.py](/https://github.com/ros-planning/moveit/blob/master/moveit_ros/visualization/src/moveit_ros_visualization/moveitjoy_module.py#L394) and retraced what's happening in there.
```
    def parseSRDF(self):
        ri = RobotInterface("/robot_description")
        planning_groups = {}
        for g in ri.get_group_names():
            self.planning_groups_tips[g] = ri.get_group_joint_tips(g)
            planning_groups[g] = ["/rviz/moveit/move_marker/goal_" + l
                                  for l in self.planning_groups_tips[g]]
        for name in planning_groups.keys():
            if len(planning_groups[name]) == 0:
                del planning_groups[name]
            else:
                print(name, planning_groups[name])
        self.planning_groups = planning_groups
        self.planning_groups_keys = planning_groups.keys()   #we'd like to store the 'order'
        self.frame_id = ri.get_planning_frame()
```
get_group_names returns just the 'manipulator' group, but ri.get_group_joint_tips returns an empty list in both setups. This leads to eventually deleting the only entry in the planning_groups dict.

```
def updatePlanningGroup(self, next_index):
        if next_index >= len(self.planning_groups_keys):
            self.current_planning_group_index = 0
```
This then evaluates to true, 0 == 0 which leads to accessing element 0 on a size 0 list.
```
next_planning_group = self.planning_groups_keys[self.current_planning_group_index]
```
Either the tips need to return something, or the parseSRDF logic is broken.
